src/lib/components/Identifiers/PIDField.js

```diff
@@ -0,0 +1,506 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2021 CERN.
+// Copyright (C) 2021 Northwestern University.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import { FastField } from 'formik';
+import _debounce from 'lodash/debounce';
+import _get from 'lodash/get';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { FieldLabel } from 'react-invenio-forms';
+import { Form, Radio } from 'semantic-ui-react';
+
+const PROVIDER_EXTERNAL = 'external';
+const UPDATE_PID_DEBOUNCE_MS = 200;
+
+/**
+ * Button component to reserve a PID.
+ */
+class BtnReservePID extends Component {
+  handleClickReservePID = () => {
+    const { onPIDReserved } = this.props;
+    //PIDS-FIXME
+    console.log('TODO implement reserve PID');
+    const value = '10.1234/as7r3-234f3';
+    onPIDReserved(value);
+  };
+
+  render() {
+    const { disabled, label } = this.props;
+    return (
+      <Form.Button
+        color="green"
+        size="mini"
+        disabled={disabled}
+        onClick={this.handleClickReservePID}
+        content={label}
+      />
+    );
+  }
+}
+
+BtnReservePID.propTypes = {
+  disabled: PropTypes.bool,
+  label: PropTypes.string.isRequired,
+  onPIDReserved: PropTypes.func.isRequired,
+};
+
+BtnReservePID.defaultProps = {
+  disabled: false,
+};
+
+/**
+ * Button component to unreserve a PID.
+ */
+class BtnUnreservePID extends Component {
```

Comment:
Same for naming if you change it above :)

Reply:
Fixed, ty!